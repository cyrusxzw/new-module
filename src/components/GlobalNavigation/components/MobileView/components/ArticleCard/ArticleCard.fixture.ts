import type { ArticleCardProps } from './ArticleCard.types';

const ArticleCardFixture: ArticleCardProps = {
  id: 'article',
  label: 'Article Card Label',
  title: 'See all Skin Care Card',
  url: '#article',
  metaLabel: 'Article Card  â€¢  Meta Label',
  image: {
    altText: 'Image of Article Card',
    sizes: {
      large:
        'https://www.aesop.com/medias/Aesop-Room-Sprays-Hybris-Cythera-Pattern-Desktop-1440x1600.jpg?context=bWFzdGVyfGltYWdlc3w0MTU4Mjl8aW1hZ2UvanBlZ3xpbWFnZXMvaGIyL2g0MS84ODEzNzIxNTUwODc4LmpwZ3wxODAxZWQ5YTdlNzMxNDYxZWZhMThiNzQ1YTJhZTRmMjE3ZWU3YzEzZDJmOGZjZWU0M2Y4ZWJlNjg2YjE1NjQ2',
      medium:
        'https://www.aesop.com/medias/Aesop-Room-Sprays-Hybris-Cythera-Pattern-Desktop-1440x1600.jpg?context=bWFzdGVyfGltYWdlc3w0MTU4Mjl8aW1hZ2UvanBlZ3xpbWFnZXMvaGIyL2g0MS84ODEzNzIxNTUwODc4LmpwZ3wxODAxZWQ5YTdlNzMxNDYxZWZhMThiNzQ1YTJhZTRmMjE3ZWU3YzEzZDJmOGZjZWU0M2Y4ZWJlNjg2YjE1NjQ2',
      small:
        'https://www.aesop.com/medias/Aesop-Room-Sprays-Hybris-Cythera-Pattern-Desktop-1440x1600.jpg?context=bWFzdGVyfGltYWdlc3w0MTU4Mjl8aW1hZ2UvanBlZ3xpbWFnZXMvaGIyL2g0MS84ODEzNzIxNTUwODc4LmpwZ3wxODAxZWQ5YTdlNzMxNDYxZWZhMThiNzQ1YTJhZTRmMjE3ZWU3YzEzZDJmOGZjZWU0M2Y4ZWJlNjg2YjE1NjQ2',
    },
  },
};

export { ArticleCardFixture };
