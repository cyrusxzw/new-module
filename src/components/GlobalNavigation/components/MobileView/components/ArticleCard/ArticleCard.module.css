@import 'styles/index';

.base {
  min-width: 80vw;
  box-sizing: border-box;
  padding-top: 16px;
  padding-bottom: 16px;
  scroll-snap-align: center;

  &.dark {
    color: var(--color-dark-label);
  }

  &.light {
    color: var(--color-light-label);
  }
}

.image {
  overflow: hidden;
  max-height: 240px;
  aspect-ratio: 16 / 9;
  pointer-events: none;

  img {
    width: 100%;
    height: auto;
    aspect-ratio: 16 / 9;
    object-fit: cover;
  }
}

.metaLabel {
  margin-top: 12px;
  font-family: var(--font-family-sans-serif-regular);
  line-height: 1.4;
}

.label {
  margin-bottom: 0;
  font-size: rem(20px);
  line-height: 1.33;
}

.leftAligned {
  flex-direction: column;
  align-items: start;
}
