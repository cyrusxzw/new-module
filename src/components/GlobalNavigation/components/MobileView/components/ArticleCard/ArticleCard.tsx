import React from 'react';
import cx from 'classnames';
import { useThemeContext } from '~/contexts';
import { Hyperlink } from '~/components/Hyperlink';
import { Paragraph } from '~/components/Paragraph';
import { Image } from '~/components/Image';
import type { ArticleCardType } from './ArticleCard.types';
import compositionStyles from '../../MobileView.module.css';
import styles from './ArticleCard.module.css';

const ArticleCard: ArticleCardType = ({
  image,
  isVisible = true,
  label,
  metaLabel,
  title,
  url,
}) => {
  const currentTheme = useThemeContext(null, 'dark');

  const classSet = cx(
    styles.base,
    styles[currentTheme],
    compositionStyles.itemElement,
    compositionStyles.ornamentalWrapper,
    styles.leftAligned,
    styles.mobileArticle,
  );

  return (
    <Hyperlink
      className={classSet}
      tabIndex={!isVisible ? -1 : null}
      theme={currentTheme}
      title={title}
      url={url}
    >
      <Image {...image} className={styles.image} />

      {metaLabel && (
        <Paragraph
          className={styles.metaLabel}
          isFlush={true}
          theme={currentTheme}
        >
          {metaLabel}
        </Paragraph>
      )}

      <Paragraph
        className={styles.label}
        hasSerifFont={true}
        isFlush={true}
        theme={currentTheme}
      >
        <span className={compositionStyles.ornamentalHover}>{label}</span>
      </Paragraph>
    </Hyperlink>
  );
};

export { ArticleCard };
