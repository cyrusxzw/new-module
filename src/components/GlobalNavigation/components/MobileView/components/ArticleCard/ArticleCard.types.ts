import type { ComponentWithoutChildren } from '~/types';
import type { Clickable } from '~/components/GlobalNavigation/GlobalNavigation.types';

type ArticleCardProps = Clickable & {
  isVisible?: boolean;
  metaLabel?: string;
  url: string;
  image?: {
    altText: string;
    sizes: {
      large?: string;
      medium?: string;
      small?: string;
      xLarge?: string;
      xSmall?: string;
    };
  };
};

type ArticleCardType = ComponentWithoutChildren<ArticleCardProps>;

export type { ArticleCardType, ArticleCardProps };
