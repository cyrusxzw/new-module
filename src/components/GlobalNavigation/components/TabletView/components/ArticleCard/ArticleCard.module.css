@import 'styles/index';

.base {
  box-sizing: border-box;
  margin-top: 15px;
  margin-bottom: 15px;

  &.dark {
    color: var(--color-dark-label);
  }

  &.light {
    color: var(--color-light-label);
  }

  &.dark:hover {
    color: var(--color-secondary);
  }

  &.light:hover {
    color: var(--color-primary);
  }
}

.image {
  overflow: hidden;
  max-height: 240px;

  /* aspect-ratio: 4 / 3; */
  aspect-ratio: 16 / 9;
  pointer-events: none;

  img {
    width: 100%;
    height: auto;

    /* aspect-ratio: 4 / 3; */
    aspect-ratio: 16 / 9;
    object-fit: cover;
  }
}

.metaLabel {
  margin-bottom: 6px;
  font-family: var(--font-family-sans-serif-regular);
  line-height: 1.4;
}

.label {
  margin-bottom: 14px;
  font-size: rem(20px);
  line-height: 1.33;
}

.leftAligned {
  flex-direction: column;
  align-items: start;
}
