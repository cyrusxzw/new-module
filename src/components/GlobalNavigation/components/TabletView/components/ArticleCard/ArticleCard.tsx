import React from 'react';
import cx from 'classnames';
import { useThemeContext } from '~/contexts';
import { Hyperlink } from '~/components/Hyperlink';
import { Image } from '~/components/Image';
import { Paragraph } from '~/components/Paragraph';
import type { ArticleCardType } from './ArticleCard.types';
import compositionStyles from '../../TabletView.module.css';
import styles from './ArticleCard.module.css';

const ArticleCard: ArticleCardType = ({
  image,
  isVisible = true,
  label,
  metaLabel,
  title,
  url,
}) => {
  const currentTheme = useThemeContext(null, 'dark');

  const classSet = cx(
    styles.base,
    styles[currentTheme],
    compositionStyles.ornamentalWrapper,
    styles.leftAligned,
  );

  return (
    <Hyperlink
      className={classSet}
      tabIndex={!isVisible ? -1 : null}
      theme={currentTheme}
      title={title}
      url={url}
    >
      {metaLabel && (
        <Paragraph
          className={styles.metaLabel}
          isFlush={true}
          theme={currentTheme}
        >
          {metaLabel}
        </Paragraph>
      )}

      <Paragraph
        className={styles.label}
        hasSerifFont={true}
        isFlush={true}
        theme={currentTheme}
      >
        <span className={compositionStyles.ornamentalHover}>{label}</span>
      </Paragraph>

      <Image {...image} className={styles.image} height="195" />
    </Hyperlink>
  );
};

export { ArticleCard };
