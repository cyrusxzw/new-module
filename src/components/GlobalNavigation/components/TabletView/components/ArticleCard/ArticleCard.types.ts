import type { ComponentWithoutChildren } from '~/types';
import type {
  Clickable,
  CategoryItemTrackingWithAction,
} from '~/components/GlobalNavigation/GlobalNavigation.types';

type ArticleCardProps = Clickable & {
  isVisible?: boolean;
  metaDuration?: string;
  metaLabel?: string;
  url: string;
  image?: {
    altText: string;
    sizes: {
      large?: string;
      medium?: string;
      small?: string;
      xLarge?: string;
      xSmall?: string;
    };
  };
};

type ArticleCardType = ComponentWithoutChildren<ArticleCardProps>;

type CategoryItemTrackingWithActionType = CategoryItemTrackingWithAction;

export type {
  ArticleCardType,
  ArticleCardProps,
  CategoryItemTrackingWithActionType,
};
