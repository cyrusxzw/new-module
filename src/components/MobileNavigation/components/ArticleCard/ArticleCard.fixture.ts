const ArticleCardFixture = {
  id: 'article-card',
  label: 'Article Card Label',
  title: 'See all Skin Care Card',
  type: 'article-card' as const,
  url: '#article-card',
  metaLabel: 'Article Card  â€¢  Meta Label',
  image: {
    altText: 'Image of Article Card',
    sizes: {
      large:
        'https://www.aesop.com/medias/Aesop-Room-Sprays-Hybris-Cythera-Pattern-Desktop-1440x1600.jpg?context=bWFzdGVyfGltYWdlc3w0MTU4Mjl8aW1hZ2UvanBlZ3xpbWFnZXMvaGIyL2g0MS84ODEzNzIxNTUwODc4LmpwZ3wxODAxZWQ5YTdlNzMxNDYxZWZhMThiNzQ1YTJhZTRmMjE3ZWU3YzEzZDJmOGZjZWU0M2Y4ZWJlNjg2YjE1NjQ2',
      medium:
        'https://www.aesop.com/medias/Aesop-Room-Sprays-Hybris-Cythera-Pattern-Desktop-1440x1600.jpg?context=bWFzdGVyfGltYWdlc3w0MTU4Mjl8aW1hZ2UvanBlZ3xpbWFnZXMvaGIyL2g0MS84ODEzNzIxNTUwODc4LmpwZ3wxODAxZWQ5YTdlNzMxNDYxZWZhMThiNzQ1YTJhZTRmMjE3ZWU3YzEzZDJmOGZjZWU0M2Y4ZWJlNjg2YjE1NjQ2',
      small:
        'https://www.aesop.com/medias/Aesop-Room-Sprays-Hybris-Cythera-Pattern-Desktop-1440x1600.jpg?context=bWFzdGVyfGltYWdlc3w0MTU4Mjl8aW1hZ2UvanBlZ3xpbWFnZXMvaGIyL2g0MS84ODEzNzIxNTUwODc4LmpwZ3wxODAxZWQ5YTdlNzMxNDYxZWZhMThiNzQ1YTJhZTRmMjE3ZWU3YzEzZDJmOGZjZWU0M2Y4ZWJlNjg2YjE1NjQ2',
    },
  },
};

export { ArticleCardFixture };
