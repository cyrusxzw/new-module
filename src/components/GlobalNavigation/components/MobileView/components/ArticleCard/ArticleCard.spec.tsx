import React from 'react';
import { render, screen } from '@testing-library/react';
import { ArticleCard } from './ArticleCard';
import { ArticleCardFixture } from './ArticleCard.fixture';

describe('<ArticleCard />', () => {
  it('should render base component correctly', () => {
    const { container } = render(
      <ArticleCard
        id={ArticleCardFixture.id}
        label={ArticleCardFixture.label}
        metaLabel={ArticleCardFixture.metaLabel}
        title={ArticleCardFixture.title}
        url={ArticleCardFixture.url}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should navigate to the article card by tab clicking', () => {
    render(
      <>
        <ArticleCard
          id={ArticleCardFixture.id}
          label={ArticleCardFixture.label}
          metaLabel={ArticleCardFixture.metaLabel}
          title={ArticleCardFixture.title}
          url={ArticleCardFixture.url}
        />
      </>,
    );

    const focusableElements = screen.getAllByRole('link');
    const one = focusableElements[0];

    expect(one).not.toHaveAttribute('tabIndex');
  });

  it('should not be able to be navigate to the article card by tab clicking', () => {
    render(
      <ArticleCard
        id={ArticleCardFixture.id}
        isVisible={false}
        label={ArticleCardFixture.label2}
        metaLabel={ArticleCardFixture.metaLabel}
        title={ArticleCardFixture.title}
        url={ArticleCardFixture.url}
      />,
    );

    const focusableElements = screen.getAllByRole('link');
    const one = focusableElements[0];

    expect(one).toHaveAttribute('tabIndex', '-1');
  });
});
