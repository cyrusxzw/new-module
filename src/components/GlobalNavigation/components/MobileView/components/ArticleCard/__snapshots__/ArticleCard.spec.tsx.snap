// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleCard /> should render base component correctly 1`] = `
<div>
  <a
    class="base left dark base dark itemElement ornamentalWrapper leftAligned"
    href="#article"
    target="_self"
    title="See all Skin Care Card"
  >
    <picture
      class="base picture dark image"
    >
      <img
        loading="lazy"
      />
    </picture>
    <p
      class="base dark flush metaLabel"
    >
      Article Card  â€¢  Meta Label
    </p>
    <p
      class="base dark flush serifFont label"
    >
      <span
        class="ornamentalHover"
      >
        Article Card Label
      </span>
    </p>
  </a>
</div>
`;
